oio/container/app.py

```diff
@@ -376,11 +380,24 @@ def generate_oio_map(self, account, container):
             map_objs.append(entry)
 
         objs = self.proxy.object_list(account, container)
-        for obj in sorted(objs['objects']):
+        for obj in sorted(objs['objects'], key=lambda x: x['name']):
             # FIXME should we backup deleted object ?
             if obj['deleted']:
                 continue
             tar = OioTarEntry(self.proxy, account, container, obj['name'])
+            if (start_block / self.CHUNK) != \
+                    ((start_block + tar.header_blocks) / self.CHUNK):
+                # header on boundary, we have to add empty block
+                padding = self.CHUNK - divmod(start_block, self.CHUNK)[1]
+                map_objs.append({
+                    'block': padding,
+                    'size': padding * BLOCKSIZE,
+                    'start_block': start_block,
+                    'slo': False,
+                    'hdr_block': padding,
+                    'end_block': start_block + padding - 1
```

Comment:
It's so confusing that I misunderstood.
* `start_block` is the index of the first block of the entry.
* `end_block` is the index of the last block of the entry.
*`hdr_block` is the number (!) of blocks required to store the header of the entry.
*`block` is the number (!) of blocks required to store the entry (including the header blocks).

Am I right?